### ISSUE TITLE
Generate plists like RoboFab’s

### ISSUE THREAD

**Issue by codingisacopingstrategy** [fontforge/fontforge#2234]
Summary, for plists:

- indent with tabs, not spaces
-  don’t indent first level children

Back in  2c7dd8e13e2b06a639e01112b796e6b8aabed8fa and 691f3a187a9762d6087072eec4f2556764d5587b FontForge changed its UFO glyph rendering to better match RoboFab’s. RoboFab being the library generating UFO’s in most other font editors, this is useful to not have noise when diffing on a project where people use multiple editing programs.

Most of the glyphs are now rendered the same, but when diffing FontForge generated additions for a Glyphs generated UFO we did run into some extra noise:

https://github.com/trevorsee/Reglo/commit/d95aa60b783ac039b5152ae1d5daabe7ce104848

I will file them in various issues. The first is that RoboFab has a different serialization for plist files. It appears to be at first glance to match the serialization used in Apple’s tools. It is defined in https://github.com/robofab-developers/robofab/blob/master/Lib/robofab/plistlib.py

Robofab and FontForge share the same preamble:

     <?xml version="1.0" encoding="UTF-8"?>
     <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
     <plist version="1.0">

Then, in RoboFab, the first level child elements of the root `plist` element are not indented. This is not exactly logical, but it matches Apple’s system generated files. The other difference is that RoboFab indents with a tab, whereas FontForge indents with two spaces.

Could FontForge use the same behaviour?
Thanks,

E

**Comment by davelab6** [comment]
Agree, and I hope @schriftgestalt can adjust Glyphs to match RoboFab whitespace

**Comment by codingisacopingstrategy** [comment]
As far as I can see Glyphs has the same whitespace as RoboFont for the plists, I imagine they all use RoboFab under the hood…

**Comment by schriftgestalt** [comment]
I was just looking into this and the example files I have use the same whitespace. The all but the glif files use tabs. I don't use RoboFab but my own implementation. I tried to follow RoboFab as close as possible.

**Comment by codingisacopingstrategy** [comment]
Thanks for your input @schriftgestalt 

Something else I noted, is that I think both Glyphs and Robofont serialize dictionaries alphabetically ordered by key.

In the example https://github.com/trevorsee/Reglo/commit/d95aa60b783ac039b5152ae1d5daabe7ce104848 this is the clearest in `fontinfo.plist`, where FontForge changes the order. Saving the same file again with RoboFont produces an alphabetically ordered fontinfo.plist.

**Comment by jvgaultney** [comment]
Yes, please do order plists alphabetically by key within dicts - even in private dicts in places like glif/lib and lib.plist. Or don't touch private lib data at all.

**Comment by davelab6** [comment]
Frank T, how many hours do you estimate this will take?

**Comment by frank-trampe** [comment]
We cannot fix this completely at the moment due to issues with Python, I think. See #1713.
